airflow/www/views.py

```diff
@@ -2720,3 +2714,36 @@ def get_count_query(self):
             .filter(models.DagModel.is_active)
             .filter(~models.DagModel.is_subdag)
         )
+
+    @has_access
+    @permission_name("list")
+    @provide_session
+    @expose('/autocomplete')
+    def autocomplete(self, session=None):
+        query = unquote(request.args.get('query', ''))
```

Comment:
Looks good -- only question is around if we need `unquote` here or not (I wouldn't have thought we do.)

Reply:
So i just tested this locally https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams does URL encoding for us so we need the unquote. Its unlikely that a dag_id would have any special character but the owner field could potentially have anything.

Reply:
I was surprised that `request.args` didn't unquote tbh. Quoting on input is 100% the behaviour we want.